### PR TITLE
use mongodb 3.6.1 client, add clustering support

### PR DESCRIPTION
**Pull request by ibmrcruicks**
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: storage/mongodb/66-mongodb.js [hunk 1]
```diff
@@ -11,13 +12,37 @@ module.exports = function(RED) {
...
+        }
+        if (this.user) {
+          url += this.user+":"+this.password+"@";
         }
-        url += this.hostname+":"+this.port+"/"+this.db;
+        if (this.clustered) {
+          url += this.hostname + "/" + this.db + "?retryWrites=true&w=majority";
+        } else {
+          url += this.hostname + ":" + this.port + "/" + this.db;
+        };
```
Comment (hardillb) [inline storage/mongodb/66-mongodb.js:43]: Can you remove the `;` from the end of this line. It is causing a test failure

File: storage/mongodb/package.json [hunk 2]
```diff
@@ -20,5 +20,11 @@
         "name": "Dave Conway-Jones",
         "email": "ceejay@vnet.ibm.com",
         "url": "http://nodered.org"
+    },
+      "contributors": [
```
Comment (hardillb) [inline storage/mongodb/package.json:24]: Can we fix the indentation here so it matches the rest

**Comment by ibmrcruicks** [comment]
done!

**Comment by dceejay** [comment]
Sorry - was away last week - How is this looking ? Still stable ? Any outstanding issues ?

**Pull request merged**